### PR TITLE
[None][fix] Report public runtime architectures in telemetry

### PR DESCRIPTION
**Pull request by Mgluhovskoi**
## Why

Encode-only endpoints can replace a checkpoint architecture such as `Qwen3ForCausalLM` with a public TensorRT-LLM runtime architecture such as `Qwen3ForTextEmbedding`. Telemetry currently reads the original checkpoint config, so it misses that effective runtime architecture. The allowlist documentation also describes every entry as a Hugging Face architecture even though public TensorRT-LLM runtime identifiers are valid privacy-reviewed entries. This also prepares the reporting path for `Qwen3ForTextReranking` in #19550.

## What changed

- Report the effective encoder runtime pretrained config, with a narrow `AttributeError` fallback to the checkpoint config so unavailable runtime metadata does not suppress the initial report and heartbeats.
- Rename `PUBLIC_HF_ARCHITECTURES` to `PUBLIC_MODEL_ARCHITECTURES` and clarify that entries may be public checkpoint or documented TensorRT-LLM runtime identifiers. Exact allowlist matches remain plaintext; unknown or custom names remain hashed.
- Keep one parameterized CPU payload-flow test covering the runtime config, generation fallback, absent executor, and missing model. It uses a stubbed executor and checks the final serialized telemetry payload.
- Add an assertion against the real executor config in the existing Qwen3 embedding GPU integration test, already included in L40S CI.

Generation models still report the checkpoint configuration. Their effective engine configuration lives in a worker process and is not directly accessible through the client LLM object, so architecture overrides supplied through `model_kwargs` are not reflected in generation telemetry. Extending reporting to that path is a follow-up.

## Validation

- Repository pre-commit hooks and Python compilation passed for the updated files.
- All four CPU payload cases passed in an isolated harness executing the checked-in reporting method and test cases with the real telemetry reporter and schema. Full llmapi pytest collection remains unverified locally because the release container encounters a TRT-LLM initialization/import failure.
- The GPU integration assertion requires model weights and GPU CI; it was not run locally.
- Earlier validation: 68 generic allowlist/privacy tests and 8 telemetry webpage architecture ingestion/aggregation tests passed.

## Dev Engineer Review
`BaseLLM._start_usage_reporting` now reports the encoder executor’s effective pretrained configuration when available. It retains `_hf_model_config` as the fallback. This enables encode-only telemetry to report architecture identifiers after runtime overrides.

The allowlist constant changes from `PUBLIC_HF_ARCHITECTURES` to `PUBLIC_MODEL_ARCHITECTURES`. The entries remain unchanged. Documentation clarifies that approved public checkpoint and documented TensorRT-LLM runtime identifiers are reported in plaintext, while other names are hashed. External consumers of the old constant name may need updates.

Review severity counts are unavailable. Test results are not established by the supplied evidence.

## QA Engineer Review
The change modifies `tests/unittest/llmapi/test_llm_telemetry.py` and `tests/unittest/usage/test_architecture_allowlist.py`. The llmapi test checks that encode-only telemetry reports `Qwen3ForTextEmbedding` with an empty hash. The allowlist test checks the renamed constant and validates its entries.

The llmapi test file appears in `tests/integration/test_lists/test-db/l0_cpu.yml`, `l0_a10.yml`, and `l0_h100.yml`. The allowlist test file was not found in the supplied test-list search results. These are unit tests, not changed integration tests. Coverage verdict: sufficient for the described encode-only reporting and allowlist behavior.

## Per-File QA Perspective
- `AGENTS.md`: Verify allowlist guidance requires public or documented identifiers and excludes private or arbitrary names.
- `README.md`: Verify telemetry documentation describes plaintext reporting for approved identifiers and hashing for other names.
- `tensorrt_llm/llmapi/llm.py`: Verify encoder runtime configuration selection and the `_hf_model_config` fallback.
- `tensorrt_llm/usage/architecture_allowlist.py`: Verify the renamed constant is used and the approved entries remain unchanged.
- `tensorrt_llm/usage/schemas/README.md`: Verify architecture extraction documentation matches effective runtime configuration behavior and empty-value handling.
- `tensorrt_llm/usage/usage_lib.py`: Verify approved names remain plaintext and other nonempty names remain hashed.
- `tests/unittest/llmapi/test_llm_telemetry.py`: Covers encode-only architecture reporting. The file is listed in the `l0_cpu`, `l0_a10`, and `l0_h100` test-db lists.
- `tests/unittest/usage/test_architecture_allowlist.py`: Covers allowlist existence and entry formatting. No test-db or manual-QA list entry was found in the supplied search results.

### CODE REVIEW

File: tensorrt_llm/llmapi/llm.py [hunk 1]
```diff
@@ -519,9 +519,13 @@ def _start_usage_reporting(self) -> None:
                     == _usage.UsageContext.UNKNOWN):
                 telemetry_config = telemetry_config.model_copy(
                     update={"usage_context": _usage.UsageContext.LLM_CLASS})
+            pretrained_config = self._hf_model_config
+            if self._encoder_executor is not None:
```
Comment (brnguyen2) [inline tensorrt_llm/llmapi/llm.py]: This four-hop chain runs inside the same `try` as `report_usage`, so an `AttributeError` anywhere in it (engine/model shape change, a future encoder executor without `.model`) drops the entire initial report and heartbeat stream for every encode-only deployment — not just the two architecture fields. `usage_lib` is otherwise carefully per-field fail-silent (`_architecture_telemetry_fields` returns `"", ""` on a bad config), and this is the one place that escalates a field problem to total telemetry loss.

Narrow it so a broken chain degrades to the checkpoint config:

```python
# Encode-only can replace the checkpoint architecture at runtime (see
# _EMBEDDING_ARCH_MAP in commands/serve.py); report what actually ran.
pretrained_config = self._hf_model_config
if getattr(self, "_encoder_executor", None) is not None:
    try:
        pretrained_config = (self._encoder_executor.model_engine.model
                             .model_config.pretrained_config)
    except AttributeError:
        pass
```

The `getattr` guard also matches `_shutdown_resources` (llm.py:1792), which already treats `_encoder_executor` as possibly absent.

File: tests/unittest/llmapi/test_llm_telemetry.py [hunk 2]
```diff
@@ -246,6 +246,62 @@ def test_telemetry_receives_hf_config_pytorch(self):
...
+class TestEffectiveRuntimeArchitectureReporting:
+    """Verify effective runtime architecture reporting end to end."""
+
+    def test_encode_only_reports_effective_runtime_architecture(
+        self, monkeypatch, enable_telemetry
+    ):
+        raw_config = SimpleNamespace(architectures=["Qwen3ForCausalLM"])
+        runtime_config = SimpleNamespace(architectures=["Qwen3ForTextEmbedding"])
+        llm = object.__new__(BaseLLM)
+        llm.args = SimpleNamespace(telemetry_config=None)
```
Comment (brnguyen2) [inline tests/unittest/llmapi/test_llm_telemetry.py]: Every hop of the chain is a `SimpleNamespace`, so this test pins the payload plumbing but nothing about the real `EncoderExecutor` → `PyTorchModelEngine` → model → `ModelConfig` shape. Combined with the swallowed exception in `_start_usage_reporting`, a rename anywhere in that chain leaves this test green while production silently falls back to the checkpoint architecture — exactly the regression the PR exists to prevent.

Worth adding an assertion against a real encoder executor in the GPU encode-only test (`tests/integration/defs/accuracy/test_llm_api_pytorch_encode.py` already builds one with `model_kwargs={"architectures": ["Qwen3ForTextEmbedding"]}`); asserting the resolved `pretrained_config.architectures[0]` there costs almost nothing and is the half this CPU test can't cover.

**Review by brnguyen2 — APPROVED** [review]
Approving — the comments below are optional touch-ups, not blockers.

Checked the premise end to end: `ModelConfig.from_pretrained` applies `model_kwargs` via `_recursive_update_config(pretrained_config, model_kwargs)` (`_torch/model_config.py:1327`), so the `Qwen3ForCausalLM` → `Qwen3ForTextEmbedding` override from `commands/serve.py:773` does land on the encoder engine's `pretrained_config`. The fix reads the right object, and the rename is safe — `architecture_allowlist` isn't in `tensorrt_llm.usage.__all__` and there are no remaining references to the old name.

One thing to state in the description: the generation path has the same gap. A user passing `model_kwargs={"architectures": [...]}` to a normal `LLM(...)` still gets the checkpoint architecture reported, because the engine lives in a worker process and isn't reachable from the `LLM` object. Saying that explicitly ("keep the HF fallback for generation models" doesn't say why) saves the next reader from assuming it was an oversight. A uniform alternative — applying `args.model_kwargs["architectures"]` to the reported value, still allowlist-gated — would cover both paths; fine as a follow-up rather than in this PR.

File: tensorrt_llm/llmapi/llm.py (hunk 1 above)
Comment (Mgluhovskoi) [reply tensorrt_llm/llmapi/llm.py]: addressed in latest commit.

File: tests/unittest/llmapi/test_llm_telemetry.py (hunk 2 above)
Comment (Mgluhovskoi) [reply tests/unittest/llmapi/test_llm_telemetry.py]: addressed in latest commit.
